### PR TITLE
feature -> develop | vaahpaypal update

### PR DESCRIPTION
**Pull request by we-himanshu-r001-v1**
Task URL : https://team.webreinvent.com/backend/collab/project/allofamicom/tasks/9

### CODE REVIEW

**Comment by themodernpk** [comment]
@we-himanshu-r001-v1 create a `Libraries/VaahPayPal.md` file and write how to use your library like [VaahStripe.md](https://github.com/webreinvent/vaahextend/blob/master/Libraries/VaahStripe.md)

File: Libraries/VaahPayPal.php [hunk 1]
```diff
@@ -0,0 +1,1014 @@
...
+    private $client_secret;
+    private $return_url;
+    private $cancel_url;
+
+    public function __construct()
+    {
+        $this->mode = env('PAYPAL_MODE');
+        $this->client_id = $this->mode == 'sandbox' ? env('PAYPAL_SANDBOX_CLIENT_ID') : env('PAYPAL_LIVE_CLIENT_ID');
+        $this->client_secret = $this->mode == 'sandbox' ? env('PAYPAL_SANDBOX_CLIENT_SECRET') : env('PAYPAL_LIVE_CLIENT_SECRET');
+        $this->return_url = url('/').'#/paypal/complete';
```
Comment (themodernpk) [inline Libraries/VaahPayPal.php]: @we-himanshu-r001-v1 is route should not be hard coded instead, you should accept a parameter in `constructor` with a default value like `url('/').'#/paypal/complete'`

Comment (themodernpk) [reply Libraries/VaahPayPal.php]: @we-himanshu-r001-v1 and route name should be a success

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: Libraries/VaahPayPal.php (hunk 1 above)
Comment (we-himanshu-r001-v1) [reply Libraries/VaahPayPal.php]: Done Please refer [435246d](https://github.com/webreinvent/vaahextend/pull/6/commits/435246dc38610c6eae126af87b7daeb4c2bf9537)

**Comment by we-himanshu-r001-v1** [comment]
> @we-himanshu-r001-v1 create a `Libraries/VaahPayPal.md` file and write how to use your library like [VaahStripe.md](https://github.com/webreinvent/vaahextend/blob/master/Libraries/VaahStripe.md)

Added : [f2e5911](https://github.com/webreinvent/vaahextend/pull/6/commits/f2e591189b2221b16e2e5b748a130d079bceab51)

File: Libraries/VaahPayPal.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+[comment]: <> ('VaahStripe' => WebReinvent\VaahExtend\Facades\VaahStripe::class,)
+
+[comment]: <> (...)
+
+[comment]: <> (])
+
+[comment]: <> (```)
+
+Add env configuration:
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md]: @we-himanshu-r001-v1 change `configuration` to `variables`

File: Libraries/VaahPayPal.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+### Dependencies
+- [PayPal-PHP-SDK](https://github.com/paypal/PayPal-PHP-SDK)
+
+### Installation
+
+```shell script
+composer require paypal/rest-api-sdk-php
+```
+
+[comment]: <> (Add Facade in `config/app.php`:)
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md]: @we-himanshu-r001-v1 this is not visible at: https://github.com/webreinvent/vaahextend/blob/f2e591189b2221b16e2e5b748a130d079bceab51/Libraries/VaahPayPal.md

Comment (themodernpk) [reply Libraries/VaahPayPal.md]: same with other lines

File: Libraries/VaahPayPal.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+[comment]: <> (...)
+
+[comment]: <> (])
+
+[comment]: <> (```)
+
+Add env configuration:
+```env
+PAYPAL_MODE=sandbox
+PAYPAL_SANDBOX_CLIENT_ID=xxxxxxxxxxxxxxx
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md:22]: @we-himanshu-r001-v1 write where and how to find `PAYPAL_SANDBOX_CLIENT_ID` and `PAYPAL_SANDBOX_CLIENT_SECRET`

File: Libraries/VaahPayPal.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+### Methods
+
+- Paypal Create Order
+
+```php
+//Initialize the VaahPaypal
+$vaahPaypal = new VaahPayPal(
+            $client_id,
+            $client_secret,
+            $return_url,
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md]: @we-himanshu-r001-v1 describe more about `$return_url` and `$cancel_url`

File: Libraries/VaahPayPal.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            $cancel_url,
+        );
+//Create Order
+$vaahPaypal->pay([
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md:74]: @we-himanshu-r001-v1 write what response will be received.

File: Libraries/VaahPayPal.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+   $vaahPaypal = new VaahPayPal(
+            $client_id,
+            $client_secret,
+            $return_url,
+            $cancel_url,
+        );
+      //Execute the order
+   $payment_id = 'xxxx';
+   $payer_id = 'xxxx';
+   $vaahPaypal->executePayment($payment_id, $payer_id);
```
Comment (themodernpk) [inline Libraries/VaahPayPal.md:108]: @we-himanshu-r001-v1 write what type of response will be received

**Review by themodernpk — CHANGES_REQUESTED** [review]

**Pull request merged**